rest_framework_gis/serializers.py

```diff
@@ -107,52 +107,103 @@ def to_representation(self, instance):
         if self.Meta.bbox_geo_field or self.Meta.auto_bbox:
             ret["bbox"] = None
 
-        for field in self.fields.values():
-            if field.write_only:
-                continue
+        id_field = self.Meta.id_field
+        geo_field = self.Meta.geo_field
+        bbox_field = self.Meta.bbox_geo_field
+        if id_field is not False and id_field in self.fields:
+            field = self.fields[id_field]
+
+            if not field.write_only:
+                value = field.get_attribute(instance)
+                ret["id"] = field.to_representation(value)
+
+        if geo_field in self.fields:
+            field = self.fields[geo_field]
+
+            if not field.write_only:
+                value = field.get_attribute(instance)
+                ret["geometry"] = field.to_representation(value)
 
-            field_name = field.field_name
-            value = field.get_attribute(instance)
-            value_repr = None
-
-            if value is not None:
-                if field_name == self.Meta.bbox_geo_field:
-                    # check for GEOSGeometry specfifc properties to generate the extent
-                    # of the geometry.
-                    if hasattr(value, 'extent'):
-                        value_repr = value.extent
-                else:
-                    value_repr = field.to_representation(value)
-
-            if self.Meta.id_field is not False and field_name == self.Meta.id_field:
-                ret["id"] = value_repr
-            elif field_name == self.Meta.geo_field:
-                ret["geometry"] = value_repr
                 if self.Meta.auto_bbox and value:
-                    ret['bbox'] = value.extent
-            elif field_name == self.Meta.bbox_geo_field:
-                ret["bbox"] = value_repr
-            elif not getattr(field, 'write_only', False):
-                ret["properties"][field_name] = value_repr
+                    ret["bbox"] = value.extent
+
+        if bbox_field in self.fields:
+            field = self.fields[bbox_field]
+
+            if not field.write_only:
+                value = field.get_attribute(instance)
+                ret["bbox"] = (value.extent if hasattr(value, 'extent') else
+                               None)
```

Comment:
generally I do not like to enforce strict line breaking because in instances like this one I have a hard time reading the code. Could you please put ``None``on the line above?